sklearn/preprocessing/label.py

```diff
@@ -154,38 +216,56 @@ def predict(self, X):
         -------
         y_hat : array, shape = (n_samples,)
             Predicted values.
+
         """
         check_is_fitted(self, "estimator_")
-        return self.inverse_func(self.estimator_.predict(X))
+        pred = self.transformer_.inverse_transform(self.estimator_.predict(X))
+        # if y is not a multi-output, it should be ravel
+        if len(self.y_shape_) == 1:
+            return pred.ravel()
+        else:
+            return pred
 
     def score(self, X, y, sample_weight=None):
-        """Computes score for regression and classification models.
+        """Returns the coefficient of determination R^2 of the prediction.
+
+        The coefficient R^2 is defined as (1 - u/v), where u is the regression
+        sum of squares ((y_true - y_pred) ** 2).sum() and v is the residual sum
+        of squares ((y_true - y_true.mean()) ** 2).sum().  Best possible score
+        is 1.0 and it can be negative (because the model can be arbitrarily
+        worse). A constant model that always predicts the expected value of y,
+        disregarding the input features, would get a R^2 score of 0.0.
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+            Test samples.
+
+        y : array-like, shape = (n_samples) or (n_samples, n_outputs)
+            True values for X.
+
+        sample_weight : array-like, shape = [n_samples], optional
+            Sample weights.
+
+        Returns
+        -------
+        score : float
+            R^2 of self.predict(X) wrt. y.
 
-        For regression, r2_score is used, for classification accuracy_score.
-        See the docstring of these functions for more details.
-        The ``r2_score`` uses variance-weighted averaging in the multi-output
-        case.
         """
 
         check_is_fitted(self, "estimator_")
-        if is_classifier(self.estimator):
-            from ..metrics import accuracy_score
-            return accuracy_score(y, self.predict(X),
-                                  sample_weight=sample_weight)
-        elif is_regressor(self.estimator):
-            from ..metrics import r2_score
-            return r2_score(y, self.predict(X), sample_weight=sample_weight,
-                            multioutput='variance_weighted')
-        else:
-            # I'm not sure if this is too many internals for an error message
+        if not is_regressor(self.estimator_):
```

Comment:
Maybe we should even remove this? I guess it prevents people from using his with a classifier and then being surprised by R2, so on second though leave it in.